### PR TITLE
Feature(create demo page) => DEV

### PR DESCRIPTION
**Pull request by sachinatcodewalnut**
# PR Template & Definition of Done

## What does this PR do?

- create a demo page for the preview

## What steps does your reviewer have to take to test this PR manually?

1. goto the /demo URL and try to check all the possible flags

## Testing checklist - Please check off

- [X] I have written unit tests for all components files I've worked on, these tests cover what the user sees and how the UI changes when they interact with it.
- [X] I have written unit tests for all helper functions I've worked on, that cover use-cases and edge-cases.
- [X] I have written integration tests for features I've worked on, to ensure my components work together as I expect.

### If you have not added tests for any of the above, please explain why below.

...

## Definition of Done - Please check off

- [X] I have completed the above testing checklist so my code is well tested and I have confidence my code works as I expect in a variety of situations.
- [X] I have Eslint and Prettier enabled to run on file save and I have fixed all errors highlighted by Eslint.
- [X] I have added complete Prop Types and type definitions without using `any` for all my components and helper functions.
- [X] I have deleted all non-descriptive comments and dead code from the files I've touched.
- [X] I have rebased my branch with the base branch I want to merge into and all the commits in this PR are my own.

### CODE REVIEW

File: src/components/molecules/DemoComponent/DemoComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            value={startDatePlaceholder}
+            onChange={(e) => setStartDatePlaceholder(e.target.value)}
+          />
+        </div>
+        <div className="mb-4">
+          <label htmlFor="endDatePlaceholder" className="block text-sm font-medium text-gray-700">
+            End Date Placeholder:
+          </label>
+          <input
+            type="text"
```
Comment (Sarath-Kumar-G) [inline src/components/molecules/DemoComponent/DemoComponent.tsx:58]: Ex component:

```JavaScript
import React, { ChangeEvent, InputHTMLAttributes, useState } from "react";
import clsx from "clsx";

export interface TextFieldProps extends InputHTMLAttributes<HTMLInputElement> {
  label?: string;
  id?: string;
  type?: string;
  additionalInputClassNames?: string;
  additionalLabelClassNames?: string;
  errorText?: string;
  value: any;
  name: string;
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
  placeholder?: string;
  tooltipText?: string;
}

export function TextField({
  label,
  id,
  errorText,
  additionalInputClassNames,
  additionalLabelClassNames,
  type = "text",
  name,
  value,
  onChange,
  placeholder,
  tooltipText,
  ...rest
}: TextFieldProps) {
  const [isTooltip, setIsTooltip] = useState<boolean>(false);

  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
    if (onChange) {
      onChange(event);
    }
  };

  const inputDynamicClassNames = clsx(additionalInputClassNames, {
    "h-12 w-full border-[1px] rounded-md rounded-tr-none pl-3 pr-6 placeholder-gray-600 focus:border-2 focus:outline-none hover:border-gray-600 focus:border-primary border-gray-200":
      true
  });

  const labelDynamicClassNames = clsx(
    additionalLabelClassNames,
    "absolute top-5 left-3 text-md flex items-center  h-[10px] hover:cursor-text",
    "transform transition-all group-focus-within:text-xs group-focus-within:-translate-y-[24px] group-focus-within:-translate-x-[4px] group-focus-within:py-0.5 group-focus-within:px-1 group-focus-within:bg-white group-focus-within:!text-primary group-focus-within:dark:bg-gray-900 group-focus-within:dark:text-white",
    {
      "-translate-y-[24px] bg-white px-1 text-xs -translate-x-[4px]": value
    }
  );

  return (
    <div>
      {label && (
        <label htmlFor={name} className={labelDynamicClassNames}>
          {label}
        </label>
      )}
      <div
        className="group flex flex-col item-center"
        onMouseEnter={() => setIsTooltip(true)}
        onMouseLeave={() => setIsTooltip(false)}
      >
        <input
          type={type}
          className={inputDynamicClassNames}
          name={name}
          onChange={handleInputChange}
          value={value}
          id={name}
          placeholder={placeholder}
          {...rest}
        />

        {errorText && (
          <span className="text-typography-error break-normal hyphens-manual text-xs pt-1 w-full ml-1">
            {errorText}
          </span>
        )}
      </div>

      {tooltipText && isTooltip && (
        <div className="absolute left-1/2 transform -translate-x-1/2 top-[72px] bg-gray-700 text-white text-[10px] rounded shadow-md z-10 text-center w-max px-2 py-1.5">
          {tooltipText}
          <div className="absolute left-1/2 transform -translate-x-1/2 -top-1 w-0 h-0 border-l-4 border-l-transparent border-r-4 border-r-transparent border-b-4 border-b-gray-700" />
        </div>
      )}
    </div>
  );
}

**Review by Sarath-Kumar-G — CHANGES_REQUESTED** [review]

File: src/components/molecules/DemoComponent/DemoComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
+import React, { useState } from "react";
+import { CustomDatePicker } from "@/components/atoms/DatePicker/DatePicker";
+
+export function DemoComponent() {
+  const [startDateLabel, setStartDateLabel] = useState("Start Date");
```
[the remark this answers is not in the document]
Comment (sachinatcodewalnut) [reply src/components/molecules/DemoComponent/DemoComponent.tsx:5]: noted;

File: src/components/molecules/DemoComponent/DemoComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            className="border rounded-md px-3 py-2 focus:outline-none focus:ring focus:border-blue-300"
+            value={startDatePlaceholder}
+            onChange={(e) => setStartDatePlaceholder(e.target.value)}
+          />
+        </div>
+        <div className="mb-4">
+          <label htmlFor="endDatePlaceholder" className="block text-sm font-medium text-gray-700">
+            End Date Placeholder:
+          </label>
+          <input
```
[the remark this answers is not in the document]
Comment (sachinatcodewalnut) [reply src/components/molecules/DemoComponent/DemoComponent.tsx]: fixed;

File: src/components/molecules/DemoComponent/DemoComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            type="text"
+            id="endDatePlaceholder"
+            className="border rounded-md px-3 py-2 focus:outline-none focus:ring focus:border-blue-300"
+            value={endDatePlaceholder}
+            onChange={(e) => setEndDatePlaceholder(e.target.value)}
```
[the remark this answers is not in the document]
Comment (sachinatcodewalnut) [reply src/components/molecules/DemoComponent/DemoComponent.tsx]: changed the name, for the date type i am not clearn ;

File: src/components/molecules/DemoComponent/DemoComponent.tsx (hunk 1 above)
Comment (sachinatcodewalnut) [reply src/components/molecules/DemoComponent/DemoComponent.tsx]: thanks, updated;

**Review by Sarath-Kumar-G — APPROVED** [review]

File: src/components/molecules/DemoComponent/DemoComponent.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Sarath-Kumar-G) [reply src/components/molecules/DemoComponent/DemoComponent.tsx]: We spoke, and resolved.

**Review by Sarath-Kumar-G — APPROVED** [review]
